tests/APIv0/SmokeTest.php

```diff
@@ -468,40 +448,153 @@ public function testPostDiscussionFromDraft() {
+            'DiscussionID' => $discussion['DiscussionID'],
+            'CommentID' => '',
+            'DraftID' => '',
+            'Format' => 'Markdown',
+            'Body' => 'Test comment draft',
+            'DeliveryType' => 'VIEW',
+            'DeliveryMethod' => 'JSON',
+            'Type' => 'Draft',
+            'LastCommentID' => 0,
+        ];
+
+        $r = $api->post("/post/comment/?discussionid={$discussionID}.json", $comment);
+        $responseCode = $r->getStatusCode();
+        $this->assertEquals(200, $responseCode);
+        $responseBody = $r->getBody();
 
         $draftModel = new \DraftModel();
-        $postedDraft = $draftModel->getWhere(['DraftID' => $responseBody['DraftID']])->firstRow(DATASET_TYPE_ARRAY);
+        $postedComment = $draftModel->getWhere(['DraftID' => $responseBody['DraftID']])->firstRow(DATASET_TYPE_ARRAY);
+        $this->assertEquals($postedComment['DiscussionID'], $comment['DiscussionID']);
+        $this->assertEquals($postedComment['Body'], $comment['Body']);
 
-        $r2 = $api->get("drafts/delete/{$postedDraft['DraftID']}/{$user['tk']}");
-        $statusCode2 = $r2->getStatusCode();
-        $this->assertEquals(200, $statusCode2);
+        return $postedComment;
     }
 
+    /**
+     * Test posting a comment draft
+     *
+     * @depends testRegisterBasic
+     * @large
+     */
+    public function testPostCommentFromDraft() {
+        $api = $this->api();
+        $api->setUser($this->getTestUser());
+
+        $draft = $this->testSavingCommentDraft();
+
+        $postComment = [
+            'DiscussionID' => $draft['DiscussionID'],
+            'CommentID' => '',
+            'DraftID' => $draft['DraftID'],
+            'Format' => 'Markdown',
+            'Body' => 'Test comment draft',
+            'DeliveryType' => 'VIEW',
+            'DeliveryMethod' => 'JSON',
+            'Type' => 'Post',
+            'LastCommentID' => 0,
+        ];
+
+        $r1 = $api->post("/post/comment/?discussionid={$draft['DiscussionID']}.json", $postComment);
+        $responseCode1 = $r1->getStatusCode();
+        $this->assertEquals(200, $responseCode1);
+        $responseBody1 = $r1->getBody();
+
+        $commentID = $responseBody1['CommentID'];
+        $r2 = $api->post("/post/comment2.json?commentid={$commentID}&inserted=1");
+        $responseCode2 = $r2->getStatusCode();
+        $this->assertEquals(200, $responseCode2);
+
+        $commentModel = new \CommentModel();
+        $dbComment = $commentModel->getWhere(['CommentID' => $commentID ])->firstRow(DATASET_TYPE_ARRAY);
+        $this->assertEquals($dbComment['DiscussionID'], $postComment['DiscussionID']);
+        $this->assertEquals($dbComment['Body'], $postComment['Body']);
+    }
+
+    /**
+     * Create a category for testing.
+     *
+     * @param string $name Category name.
+     * @param  string $url Category url.
+     * @return array $category
+     */
+    private function createCategory($name = null, $url = null) {
```

Comment:
I would've also accepted a dumb static int var in here that's incremented whenever the method is called and uses that value to create new names/URL codes.